src/app/applications/applications-list/applications-table/applications-table.component.ts

```diff
@@ -110,10 +122,42 @@ export class ApplicationsTableComponent implements AfterViewInit, OnInit {
   }
 
   applicationHasDevices(id: number): boolean {
-    const applicationToDelete = this.data?.find(app => app.id === id);
+    const applicationToDelete = this.data?.find((app) => app.id === id);
     return applicationToDelete && applicationToDelete.iotDevices.length > 0;
   }
 
+  applicationHasSigFoxDevices(id: number): boolean {
+    const applicationToDelete = this.data?.find((app) => app.id === id);
+    const checkForSigfox = applicationToDelete.iotDevices.find((device) => {
+      return device.type === DeviceType.SIGFOX;
+    });
+    if (checkForSigfox) {
+      return true;
+    } else return false;
+  }
+
+  canBeDeleted(id: number): boolean {
```

Comment:
From the name I would expect that this just checks if the application can the deleted - not also show a prompt. I actually don't even think this method is necessary - let's have a talk about it